lessons/python_basics/practice_code.py

```diff
@@ -0,0 +1,277 @@
+#They're easier on memory and processor than lists.
+
+mytuple = (1, "2", 3j, 4.000, "five")
+
+#Python also has dictionaries, which are unordered sets of values accessible 
+#by string keys.
+
+mydict = {"dogs": "DOGS!", "dognumber": 47}			#Can also just write {} for a blank dictionary
+mydict["dognames"] = ["spot", "rover", "agamemnon"]	#Can add dictionary entries after initialization
+print mydict										#Print is versatile, and will give you a summary of your dict
+del mydict["dogs"]									#Delete a key and its associated value
+
+#Note that passing an "immutable type" of object, like a float, tuple 
+#or int, using "=" will make a copy of the value.
+
+x = 4
+y = x
+print x, y
+y = 7
+print x, y
+
+#Lists and dicts, however, are "mutable", and 
+
+x = ["alpha", "beta"]
+y = x
+print x, y
+y[1] = "gamma"
+print x, y
+
+#To make a copy of a list, do:
+
+y = list(x)		#x[:] also works
+y[1] = "delta"
+print x, y
+
+#See https://en.wikibooks.org/wiki/Python_Programming/Data_Types#Mutable_vs_Immutable_Objects
+
+
+########## IMPORTING LIBRARIES ##########
+
+#If you want Matlab-style arrays, you'll need to install the 
+#numpy library.  Here, "try/except" means try <try code>, and 
+#if Python throws up an exception, do <except code>.
+
+try:
+    import numpy as np						#Imports numpy library
+    myarray = np.array([1,43,32.1,17,216])	#Create numpy array
+    print myarray
+    print myarray*myarray/3.				#Multiplication is element by element; "3." is typecast to an array of length myarray
+except:
+    print "You don't have numpy installed!  This section of code won't run."
+
+
+
+########## LOOPS & CONDITIONALS ##########
+
+#Like the try/except above, conditional statements and loops in 
+#Python are demarcated using
+
+#For loop that cycles over range(starting number, end number, 
+#stepsize).
+
+for i in range(3,10,2):
+    print i
+
+#For loop; cycles over all elements in mylist from previous section.
+
+for item in mylist:
+    print item
+
+#For loop that searches mylist for "five".
+
+for item in mylist:
+    if item == "five":
+        print "'five' found!"
+        break					#Exits loop
+    else:
+        print "Not yet..."
+
+#The standard way of indenting in Python is four spaces.  
+#Tabs also work (I use them).
+
+i = 0
+while i < 10:					#While loop
+    if i > 5 and not i > 8:
+        print i*i
+    i += 1
+
+
+########## FUNCTION DECLARATION ##########
+
+#Here's a simple function.  (str() typecasts to string.)
+
+def myf(x,y):
+    print 'You called myf(x,y) with the value x = ' + str(x) + ' and y = ' + str(y)
+    print 'x * y = ' + str(x*y)
+
+myf(13,17)
+
+#Functions don't have to be declared in a header, and arguments 
+#don't need to have a pre-declared data type.
+
+def printtype(x="dummy"):
+    print "You've entered a ", type(x), "whose value is", x
+    return type(x)
+
+rt = printtype(4)
+rt = printtype("eight")
+rt = printtype()
+print rt
+
+#Here I've also given argument "x" a default value; if x is 
```

Comment:
I think you should move this comment closer to the function. Maybe even make it a docstring.